### ISSUE TITLE
Slide background overlap in the pdf export

### ISSUE THREAD

**Issue by saimn** [hakimel/reveal.js#1261]
Following #1252, if I use an empty slide with a background image (to show a fullscreen image), the background image overlaps with the next slide content in the pdf export:
```
<section data-background="https://s3.amazonaws.com/hakim-static/reveal-js/image-placeholder.png">
</section>
<section data-background="https://s3.amazonaws.com/hakim-static/reveal-js/image-placeholder.png" data-background-repeat="repeat" data-background-size="100px">
....
```

![2015-06-11-103436_1680x1050_scrot](https://cloud.githubusercontent.com/assets/311639/8103052/aff0b402-1025-11e5-9198-9dfa175b3c5c.png)

This can be avoided by adding some fake content in the empty section (`<div>&nbsp;</div>`), but it can be surprising when discovering reveal.js

**Comment by ddeboer** [comment]
I’m having this (or a similar) problem too. You can see that (black) background color from the previous slides slightly overflows to the next (grey) slide.

![screen shot 2015-06-28 at 21 14 47](https://cloud.githubusercontent.com/assets/89267/8397869/be7f0d58-1dda-11e5-83d7-9bdac3cdace1.png)

@saimn There seem to be no empty sections in the above slides. What exactly do you mean with adding the empty div?

**Comment by obilodeau** [comment]
I was bitten by both of your PDF rendering bugs today. I looked at the code and tried to fix them but it seems that even though I triggered the `@media print` in the developer console I couldn't really emulate the print view and inspect how CSS affected the HTML tags and interactively debug it.

I could get a proper render with phantomjs and the [print-pdf phantomjs plugin](https://github.com/hakimel/reveal.js/blob/master/plugin/print-pdf/print-pdf.js) but links weren't working...

Long story short, I found about [decktape](https://github.com/astefanutti/decktape) which instruments phantomjs to crawl the slides and dump them into a PDF and it fixed all of my problems.

I'll open a PR to suggest trying it in the README under the print-pdf option.

**Comment by ddeboer** [comment]
Thanks @obilodeau, using decktape worked for me!

**Comment by opensource21** [comment]
decktape has a different focus. It creates a page for each fragment which is shown on a slide. More painful I wasn't able to create output in the format of DIN A4. So it would be nice if the issue could be fixed.

**Comment by jdwolin** [comment]
Having this same issue of overlapping slides on PDF export even with content.  @obilodeau, can you explain more specifically how you got decktape to work in place of the default PDF export package?

**Comment by opensource21** [comment]
I start the presentation with Bracket and call
`decktape bin\phantomjs decktape.js --size=842x595 http://127.0.0.1:51824/index.html mypresentation.pdf`
Hope this makes it clear. Perhaps you can use `file://` too, but I didn't make a test.

**Comment by scor** [comment]
confirmed using ```file://``` works with decktape for local slides. had similar issues with the default export plugin where links were not exported as links in the PDF. decktape worked like a charm.

**Comment by DaveEveritt** [comment]
It *is* an issue, but so easily fixed with the blank &nbsp; content, it's not a deal-breaker.

**Comment by obilodeau** [comment]
@DaveEveritt can you elaborate? I don't understand your workaround.

**Comment by DaveEveritt** [comment]
See the first post: "This can be avoided by adding some fake content in the empty section" i.e. this is (simplified) from my slides:
`<section title="screenshot of the Guardian" data-background-image="images/guardian.png" data-background-size="900px"><div>&nbsp;</div></section>`

in fact, the extra div is excessive:
`<section title="screenshot of the Guardian" data-background-image="images/guardian.png" data-background-size="900px">&nbsp;</section>`

The other gotcha is that the preview in the browser will still have bad overlaps (not just on the background-image pages), but when you hit 'Print' and save as PDF, pages that contain only a background image will appear correctly.

**Comment by jiayihu** [comment]
I was having the same issue and I can confirm that @DaveEveritt suggestion works perfectly. Maybe the print script should add the node in `print-pdf` mode.

**Comment by obilodeau** [comment]
The work-around does work @DaveEveritt. Thanks!

There is only one problem. I use some CSS to do text positioning on the title slide using `margin-top` specifying sizes in `em` and the "background drift" problem happens again. The CSS should only apply to h1, h2 elements and not the canvas so I don't understand what's going on.

Also, I'm not good enough in `@media` CSS to understand what I should do to fix this so that my positioning would work on both screen and print.

**Comment by DaveEveritt** [comment]
obilodeau - AFAIK there's a separate print stylesheet where you could - presumably - adjust your added styles.
